core/src/main/scala/cilib/Archive.scala

```diff
@@ -0,0 +1,94 @@
+package cilib
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric.Positive
+import scalaz.Scalaz._
+import scalaz._
+
+sealed trait ArchiveBound
+final case class Bounded[A](limit: Int Refined Positive, deletePolicy: List[A] => A) extends ArchiveBound
+final case class Unbounded() extends ArchiveBound
+
+sealed abstract class Archive[A] {
+  import Archive._
+
+  def values: List[A] =
+    this match {
+      case Empty(_, _)       => List()
+      case NonEmpty(l, _, _) => l
+    }
+
+  def bound: ArchiveBound =
+    this match {
+      case Empty(b, _)       => b
+      case NonEmpty(_, b, _) => b
+    }
+
+  def insertCondition: (A, A) => Boolean =
```

Comment:
I'm not sure if this function will ever be used - remove it possibly?

Reply:
@gpampara I don't see a use for it at the moment so will remove it. I will just have to confirm with @KyleErwin.